src/LinAlg/hiopVectorRajaPar.hpp

```diff
@@ -153,7 +153,10 @@ class hiopVectorRajaPar : public hiopVector
   virtual void copyToStarting(int start_index_in_src, hiopVector& v) const;
   /* Copy 'this' to v starting at start_index in 'v'. */
   virtual void copyToStarting(hiopVector& v, int start_index_in_dest) const;
-  virtual void copyToStartingAt_w_pattern(hiopVector& v, int start_index_in_dest, const hiopVector& ix) const;
+  virtual void copyToStartingAt_w_pattern(hiopVector& v,
+                                          int start_index_in_dest,
+                                          const hiopVector& ix,
+                                          hiopVectorInt* idx_map = nullptr) const;
```

Comment:
I think having `idx_map` as a member in the vector class would offer better encapsulation. A good name would be `idx_cumsum_`

Reply:
the reason I didn't implemented it like the way you mentioned, is that `idx_cumsum_` may need to be updated in many functions. For example, in `copyTo` and `copyFrom`. See the latest commit for example. (note that I haven't reset `idx_map` in all the functions yet)